artemis/placeholder_page_detector.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any
+
+import requests
+
+from artemis import http_requests
+from artemis.config import Config
+
+
+class PlaceholderPageAnalyzer:
+    def __init__(self, url: str) -> None:
+        self.url = url
+
+    def check_response(self) -> Any:
+        if self.url.startswith(("https://", "http://")):
+            try:
+                response = http_requests.get(self.url)
+            except requests.RequestException:
+                return False
+        else:
+            url = "http://" + self.url
+            try:
+                response = http_requests.get(url)
+            except requests.RequestException:
+                url = "https://" + self.url
+                try:
+                    response = http_requests.get(url)
+                except requests.RequestException:
+                    return False
+        response.encoding = "utf-8"
+        return response
+
+    def is_placeholder(self) -> bool:
+        response = self.check_response()
+        if response:
+            html_content = response.content
+            with open(
+                Config.Modules.PlaceholderPageContent.PLACEHOLDER_PAGE_CONTENT_FILENAME, "r", encoding="utf-8"
```

Comment:
this will cause the file to be read on every check, this can be made more optimal by moving to __init__ and constrycting a singleton instance (and calling is_placeholder on different urls)